src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -265,12 +265,6 @@ internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diag
 
             foreach (var diag in supportedDiagnostics)
             {
-                if (HasNotConfigurableTag(diag.CustomTags))
```

Comment:
so shouldn't the engine check whether diagnostic has NotConfiguration tag, and if it has, make sure it never ask things like compiler options to get overridden value?